sunshine/httpcommon.cpp

```diff
@@ -180,4 +181,52 @@ int create_creds(const std::string &pkey, const std::string &cert) {
 
   return 0;
 }
+
+bool download_file(const std::string &url, const std::string &file) {
+  CURL *curl = curl_easy_init();
+  if(!curl) {
+    BOOST_LOG(error) << "Couldn't create CURL instance";
+    return false;
+  }
+  FILE *fp = fopen(file.c_str(), "wb");
+  if(!fp) {
+    BOOST_LOG(error) << "Couldn't open ["sv << file << ']';
+    curl_easy_cleanup(curl);
+    return false;
+  }
+  curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+  curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, fwrite);
+  curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
+  CURLcode result = curl_easy_perform(curl);
+  if(result != CURLE_OK) {
+    BOOST_LOG(error) << "Couldn't download ["sv << url << ", code:" << result << ']';
+  }
```

Comment:
Revisiting the original error about not being able to upload the cover.

Terminal output
```txt
[2022:08:20:18:36:42]: Error: Couldn't download [https://images.igdb.com/igdb/image/upload/t_cover_big_2x/co1rg8.png, code:77]
```

The `/assets/covers/` directory is created, and files are created inside the directory, but they are of 0 size.

![image](https://user-images.githubusercontent.com/42013603/185768547-ee9f3a86-6f3b-470b-8829-65119f068e0c.png)

I looked up code 77 and found this:
```
CURLE_SSL_CACERT_BADFILE (77)

Problem with reading the SSL CA cert (path? access rights?) 
```

I suppose it's an issue with the sunshine self signed certificate.

Reply:
I think this has nothing to do with self-signed certificate. But something is wrong with the certificate storage in your system.

Reply:
Okay, I have no idea what cert curl is complaining about or where it expects to find it.

I'm testing in Windows. Sunshine does not store certs in the windows certificate store (or whatever it's called), just inside the install directory.

I get no warnings when navigating to the image url, only when navigating to sunshine's ui, due to the cert being self signed. You have to accept a warning in the browser to proceed. So figured it was related to that.

Reply:
@mariotaku are you using a fresh instance of sunshine when testing, and do you use the certificate that is created by sunshine?

This is all I have tried.

- Reset certs in Firefox (issue is not browser dependent, was using Chrome incognito previously)
- Added sunshine cert pem file to Windows "Trusted Root Certification Authorities"
- Tried from a browser on another device instead of the host
- Was also going to test from Windows 10 but cannot start this build of sunshine on either of my Windows 10 systems (updated comment above).

Not really sure what else to try on my end.

Reply:
According to some docs, that's because openssl needs to be included, too. I'll take a look when I got time.